app/src/main/java/org/thoughtcrime/securesms/contacts/avatars/ContactColorsLegacy.java

```diff
@@ -0,0 +1,63 @@
+package org.thoughtcrime.securesms.contacts.avatars;
+
+import androidx.annotation.NonNull;
+
+import org.thoughtcrime.securesms.color.MaterialColor;
+
+/**
+ * Used for migrating legacy colors to modern colors. For normal color generation, use
+ * {@link ContactColors}.
+ */
+public class ContactColorsLegacy {
+
+  private static final String[] LEGACY_PALETTE = new String[] {
+      "red",
+      "pink",
+      "purple",
+      "deep_purple",
+      "indigo",
+      "blue",
+      "light_blue",
+      "cyan",
+      "teal",
+      "green",
+      "light_green",
+      "orange",
+      "deep_orange",
+      "amber",
+      "blue_grey"
+  };
+
+  private static final String[] LEGACY_PALETTE_2 = new String[]{
+      "pink",
+      "red",
+      "orange",
+      "purple",
+      "blue",
+      "indigo",
+      "green",
+      "light_green",
+      "teal",
+      "brown",
+      "blue_grey"
+  };
+
+
+  public static MaterialColor generateFor(@NonNull String name) {
+    String serialized = LEGACY_PALETTE[Math.abs(name.hashCode()) % LEGACY_PALETTE.length];
+    try {
+      return MaterialColor.fromSerialized(serialized);
+    } catch (MaterialColor.UnknownColorException e) {
+      return ContactColors.generateFor(name);
+    }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 47:52, 56:61.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.